pkg/util/validation/limits.go

```diff
@@ -557,12 +563,15 @@ func (l *Limits) MarshalYAML() (interface{}, error) {
 }
 
 func (l *Limits) validate() error {
+	if model.ValidationScheme(l.NameValidationScheme) == model.UnsetValidation {
+		l.NameValidationScheme = ValidationSchemeFlag(model.LegacyValidation)
+	}
```

Comment:
I think the following is safer:
```suggestion
	switch model.ValidationScheme(l.NameValidationScheme) {
	case model.UTF8Validation, model.LegacyValidation:
	case model.UnsetValidation:
		l.NameValidationScheme = ValidationSchemeValue(model.LegacyValidation)
	default:
		return fmt.Errorf("unrecognized name validation scheme: %s", l.NameValidationScheme)
	}
```